app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt

```diff
@@ -186,6 +232,72 @@ class MapViewModel : ViewModel() {
     }
   }
 
+  /**
+   * Draw markers on the map.
+   *
+   * @param pointAnnotationManager the PointAnnotationManager to draw the markers on
+   * @param parkingList the list of parkings to draw
+   * @param bitmap the bitmap to use for the markers
+   */
+  fun drawMarkers(
+      pointAnnotationManager: PointAnnotationManager?,
+      parkingList: List<Parking>,
+      bitmap: Bitmap,
+  ) {
+    parkingList.forEach {
+      pointAnnotationManager?.create(
+          PointAnnotationOptions()
+              .withPoint(it.location.center)
+              .withIconImage(bitmap)
+              .withIconAnchor(IconAnchor.BOTTOM)
+              .withIconOffset(listOf(0.0, bitmap.height / 12.0))
+              .withData(Gson().toJsonTree(it)))
+    }
+  }
+
+  /**
+   * Draw the rectangles on the map
+   *
+   * @param polygonAnnotationManager the polygon annotation manager
+   * @param parkingsList the list of locations to draw
+   */
+  fun drawRectangles(
+      polygonAnnotationManager: PolygonAnnotationManager?,
+      plabelAnnotationManager: PointAnnotationManager?,
+      parkingsList: List<Parking>
+  ) {
+    val annotations: MutableList<PolygonAnnotationOptions> = mutableListOf()
```

Comment:
Some inline comments here would be nice, it's a little hard to follow

Reply:
Done. Tell me if that works for you.